### PR TITLE
feat: add modal slots

### PR DESCRIPTION
**Pull request by saraparsa13**

### CODE REVIEW

File: src/modal/modal.ts [hunk 1]
```diff
@@ -6,10 +6,21 @@ export class Modal extends LitElement implements HTMLDialogElement {
...
   open: boolean = false;
 
+  @property({ type: String })
+  title: string = "";
+
+  @property({ type: String })
+  description: string = "";
+
+  @property({ type: String })
+  alignment: string = 'right';
```
Comment (amir78729) [inline src/modal/modal.ts:19]: Can we make the type of this property more specific using union operation? for example `'left' | 'center' | 'right'`

File: src/modal/modal.stories.ts [hunk 2]
```diff
@@ -1,12 +1,28 @@
 import { html, TemplateResult } from 'lit';
-import './index.js';
 import { Meta } from '@storybook/web-components';
+import './index.js';
+import '@tapsioss/icons/dist/icons/circle-check-fill'
```
Comment (amir78729) [inline src/modal/modal.stories.ts:4]: Please apply the ESLint fix command to this file

```suggestion
import '@tapsioss/icons/dist/icons/circle-check-fill';
```

File: src/modal/modal.stories.ts [hunk 3]
```diff
@@ -18,15 +34,64 @@ interface Story<T> {
...
+  alignment: string;
+  layout: string;
+  leadingType: string;
+}
+
+const renderActions = (layout: string) => {
+  if (layout === 'vertical') {
+    return html`
+      <div slot="actions" style="display: flex; flex-direction: column; gap: 12px">
+        <tap-button tabindex="0" style="width:100%">عنوان دکمه</tap-button>
```
Comment (amir78729) [inline src/modal/modal.stories.ts:46]: Shouldn't we import the button component in this file?

File: src/modal/modal.ts [hunk 4]
```diff
@@ -114,7 +125,21 @@ export class Modal extends LitElement implements HTMLDialogElement {
...
         >
-          <slot> </slot>
+        ${this.image
+        ? html`
+          <div class="image-container" part="image-container">
+            <img class="image" part="image" src=${this.image} alt="dialog image" />
+          </div>`
+        : html`
+          <div class="icon" part="icon"><slot name="icon"></slot></div>
+        `}
```
Comment (majidsajadi) [inline src/modal/modal.ts:135]: what if we dont have neither `image-container` nor `icon` ?

Comment (majidsajadi) [inline src/modal/modal.ts:135]: why do we have `image` as attribute but `icon` as slot?

**Pull request merged**